src/mobu/services/business/notebookrunner.py

```diff
@@ -316,42 +324,63 @@ async def execute_code(self, session: JupyterLabSession) -> None:
                 return
 
             self._notebook = self.next_notebook()
-
+            relative_notebook = str(
+                self._notebook.relative_to(self._repo_dir or "/")
+            )
             iteration = f"{count + 1}/{num_executions}"
             msg = f"Notebook {self._notebook.name} iteration {iteration}"
             self.logger.info(msg)
 
-            with self.timings.start(
-                "execute_notebook", self.annotations(self._notebook.name)
-            ) as sw:
-                try:
-                    for cell in self.read_notebook(self._notebook):
-                        code = "".join(cell["source"])
-                        cell_id = cell.get("id") or cell["_index"]
-                        ctx = CodeContext(
-                            notebook=self._notebook.name,
-                            path=str(self._notebook),
-                            cell=cell_id,
-                            cell_number=f"#{cell['_index']}",
-                            cell_source=code,
+            set_tag("notebook", relative_notebook)
+            notebook_info = {
+                "notebook": relative_notebook,
+                "iteration": iteration,
+            }
+            set_context("notebook_info", notebook_info)
+
+            # We need to start a span around this transaction becaues if we
+            # don't, the nested transaction shows up as "No instrumentation" in
+            # the enclosing transaction in the Sentry UI.
+            with start_span(
+                name="execute_notebook", op="execute_notebook"
+            ) as notebook_span:
+                notebook_span.set_data("notebook_info", notebook_info)
+                with start_transaction(
+                    name="execute_notebook",
+                    op="execute_notebook",
+                ) as span:
```

Comment:
This code has wandered into a really high indentation level. I think it would be worth refactoring this into helper methods for clarity.